tracing/tracing.go

```diff
@@ -33,9 +39,32 @@ func defaultConfig() *config {
 		serviceName: envOrDefault(signalfxServiceName),
 		accessToken: envOrDefault(signalfxAccessToken),
 		url:         envOrDefault(signalfxEndpointURL),
+		globalTags:  envGlobalTags(),
+
 	}
 }
 
+// envGlobalTags extract global tags from the environment variable and parses the value in the expected format
+// key1:value1,
+func envGlobalTags() []tracer.StartOption {
+	var globalTags []tracer.StartOption
+	if val := os.Getenv(signalfxSpanTags); val != "" {
+		tags := strings.Split(val, ",")
+		for _, tag := range tags {
+			pair :=strings.Split(tag, ":")
+			if len(pair) == 2 {
+				key := strings.TrimSpace(pair[0])
+				value := strings.TrimSpace(pair[1])
+				if key != "" && value != "" {
```

Comment:
Is it really necessary to enforce non-empty values?

Reply:
Yes - in testing if the value was empty it would add an entry to the map with the key and the empty value.

Reply:
yes, but having a key with an empty value is different from the key not being present...

Reply:
Doesn't it just seem like bad practice to have an empty value?

Reply:
Some tracers will remove existing tags if they are set w/ an empty value.  We definitely shouldn't be using them in zipkin encoding at the least, so this is a good todo.

Reply:
This is not OTel but fwiw the [OTel spec](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#set-attributes) says: "Attribute values expressing a numerical value of zero or an empty string are considered meaningful and MUST be stored and passed on to span processors/exporters. Attribute values of null are considered to be not set and get discarded as if that SetAttribute call had never been made."

Reply:
Hmm good to know about the spec.
@rmfitzpatrick - do you know which tracers remove tags that are set with empty values?

Reply:
oh - so i found this in the zipkin api https://github.com/openzipkin/zipkin-api/blob/master/zipkin-jsonv2.proto#L50-L51
it says that empty values are valid but empty keys are not. Since this is only for the SFx tracer, i'm going to remove the check for value being empty.